src/error.rs

```diff
@@ -310,5 +324,22 @@ impl From<LzwError> for TiffError {
     }
 }
 
+impl From<jpeg::Error> for TiffError {
+    fn from(error: jpeg::Error) -> Self {
+        match error {
+            jpeg::Error::Format(format) => {
+                TiffError::FormatError(TiffFormatError::Format(format!("JPEG format: {}", format)))
+            }
+            jpeg::Error::Unsupported(unsupported) => TiffError::UnsupportedError(
+                TiffUnsupportedError::UnsupportedJpegFeature(unsupported),
+            ),
+            jpeg::Error::Io(io_error) => TiffError::IoError(io_error),
+            jpeg::Error::Internal(internal) => {
+                panic!("According to jpeg create, this is not used, so this shouldn't be reached..")
```

Comment:
We should still report it, waiting on the version bump to actually remove the variant. Let's not rely on what looks like an implementation detail here. Let's report it as a `FormatError` with some opaque wrapper around the jpeg error. Actually, it would make sense to have that wrapper for both this and the `jpeg::Error::Format` variant without further discriminating between the two.

```rust
#[derive(Debug)]
struct JpegDecoderError {
    inner: jpeg::Error,
}

// and a display impl to replace eager `format!("JPEG format:  {}", error)`
// Note: Display for jpeg::Error already includes "invalid JPEG format: {}"
```

Reply:
Do you mean like the version I just pushed?

If so, this would require implementing `Clone` and `PartialEq` on `jpeg::Error` to be able to be merged.

Reply:
Mostly, `PartialEq` can always be added manually, `Clone` could be solved by holding the error within an `Arc` if necessary. Does that sound like an adequate solution? I'm not entirely certain.

Reply:
I think it is probably fine, and can always be cleaned up later if desired. Latest push should include your changes and be mergable without changes to jpeg-decoder